### ISSUE TITLE
Arg types for AtenBatchNormOp

### ISSUE THREAD

**Issue by anupgangwar** [llvm/torch-mlir#540]
Hi,
The test 'BatchNorm1DModule_basic' creates the following Torchscript MLIR:

```
module attributes {torch.debug_module_name = "BatchNorm1DModule"}  {
  func @forward(%arg0: !torch.vtensor<[10,4,3],f32> ) -> !torch.vtensor<[10,4,3],f32> {
    %0 = torch.vtensor.literal(dense<[5.000000e-01, 4.000000e-01, 3.000000e-01, 6.000000e-01]> : tensor<4xf32>) : !torch.vtensor<[4],f32>
    %1 = torch.vtensor.literal(dense<[3.000000e+00, 2.000000e+00, 4.000000e+00, 5.000000e+00]> : tensor<4xf32>) : !torch.vtensor<[4],f32>
    %float1.000000e-01 = torch.constant.float 1.000000e-01
    %float1.000000e-05 = torch.constant.float 1.000000e-05
    %true = torch.constant.bool true
    %false = torch.constant.bool false
    %2 = torch.aten.batch_norm %arg0, %1, %0, %0, %1, %false, %float1.000000e-01, %float1.000000e-05, %true : !torch.vtensor<[10,4,3],f32>, !torch.vtensor<[4],f32>, !torch.vtensor<[4],f32>, !torch.vtensor<[4],f32>, !torch.vtensor<[4],f32>, !torch.bool, !torch.float, !torch.float, !torch.bool -> !torch.vtensor<[10,4,3],f32>
    return %2 : !torch.vtensor<[10,4,3],f32>
  }
}
```

Which means that input is of type tensor<10x4x3xf32> and mean is of type tensor<4xf32>. My understanding is that the mean should have the same rank as the input. Is that correct?

**Comment by anupgangwar** [comment]
Tagging @cathyzhyi and @sjarus.

**Comment by cathyzhyi** [comment]
The mean and standard-deviation are calculated over the mini-batches per feature see https://pytorch.org/docs/stable/generated/torch.nn.BatchNorm1d.html. Therefore, mean and std are 1D tensors with size of number of features. For input of N,C,H,W mean's sizes should be 1xC. You can also verify this with snippets in pytorch
```
>>> m = torch.nn.BatchNorm2d(100)
>>> m.running_mean.size()
torch.Size([100])
```
